avaframe/ana5Utils/DFAPathGeneration.py

```diff
@@ -0,0 +1,486 @@
+        # we want the origin to be in (0, 0) as it is in the avaProfile that comes in
+        X, Y = gT.makeCoordinateGrid(0, 0, csz, ncols, nrows)
+        indNonZero = np.where(fieldsList[0]['FD'] > 0)
+        # convert this data in a particles style (dict with x, y, z info)
+        particlesIni = {'x': X[indNonZero], 'y': Y[indNonZero]}
+        particlesIni, _ = gT.projectOnRaster(dem, particlesIni)
+        log.info('Using fields to generate avalanche path profile')
+
+    avaProfileExt = extendDFAPathKernel(cfg, avaProfile, dem, particlesIni)
+    return avaProfileExt
+
+
+def extendDFAPathKernel(cfg, avaProfile, dem, particlesIni):
+    """ extend the DFA path at the top and bottom
+    avaProfile with x, y, z, s information
+
+    Parameters
+    -----------
+    cfg: configParser
+        configuration object with:
+        extTopOption: int
+        how to extend towards the top?
+            0 for heighst point method
+            a for largest runout method
+        nCellsResample: int
+            resampling length is given by nCellsResample*demCellSize
+        nCellsMinExtend: int
+            when extending towards the bottom, take points at more than nCellsMinExtend*demCellSize
+            from last point to get the direction
+        nCellsMaxExtend: int
+            when extending towards the bottom, take points at less than nCellsMaxExtend*demCellSize
+            from last point to get the direction
+        factBottomExt: float
+            extend the profile from factBottomExt*sMax
+    avaProfile: dict
+        profile to be extended
+    dem: dict
+        dem dict
+    particlesIni: dict
+        initial particles dict
+
+    Returns
+    --------
+    avaProfileExt: dict
+        extended profile at top and bottom (x, y, z).
+    """
+    # resample the profile
+    resampleDistance = cfg.getfloat('nCellsResample') * dem['header']['cellsize']
+    avaProfile = extendProfileTop(cfg.getint('extTopOption'), particlesIni, avaProfile)
+    avaProfile, _ =  gT.prepareLine(dem, avaProfile, distance=resampleDistance, Point=None)
+    avaProfile = extendProfileBottom(cfg, dem, avaProfile)
+    avaProfile, _ =  gT.prepareLine(dem, avaProfile, distance=resampleDistance, Point=None)
+    # project the profile on the dem
+    avaProfile, _ =  gT.projectOnRaster(dem, avaProfile, interp='bilinear')
+    # remove points that lay outside of the dem
+    isNotNan = ~(np.isnan(avaProfile['z']))
+    avaProfileExt = {}
+    avaProfileExt['x'] = avaProfile['x'][isNotNan]
+    avaProfileExt['y'] = avaProfile['y'][isNotNan]
+    avaProfileExt['z'] = avaProfile['z'][isNotNan]
+    return avaProfileExt
+
+
+def extendProfileTop(extTopOption, particlesIni, profile):
+    """ extend the DFA path at the top (release)
+
+    Either towards the highest point in particlesIni (extTopOption = 0)
+    or the point leading to the longest runout (extTopOption = 1)
+
+    Parameters
+    -----------
+    extTopOption: boolean
```

Comment:
In the ini extTopOption is not boolean but an integer, with potentially more options, right?

And then I suggest catching if `extTopOption` is something else than 0 or 1 in an else (with an error)